src/assets/RestApi/rest-api.md

```diff
@@ -0,0 +1,400 @@
+```
+- `@app.route` binds a URL pattern `/tasks/<int:task_id>` to the function that follows it.
+- `<int:task_id>` is a variable part of the URL that will be captured and passed to the function as an argument.
+- `methods=["PUT"]` specifies that this route should respond to the HTTP PUT requests. This means that when a client sends a PUT request to the `/tasks/<int:task_id>` URL, Flask will call the associated function.
+- `update_task()` function will take `task_id` as the argument that is extracted from `/tasks/<int:task_id>` URL. The function will then search for a task with that ID in the tasks list. If a task is found, it will update the task with the new data provided in the request and return the updated task in JSON format. If no task is found, it will return a `404` status code with an error message.
+- `task.update(request.get_json())` updates the task with the JSON data sent by the client in the PUT request. The `request.get_json()` function parses the incoming request data as JSON, and the `update()` method merges this data into the `task` dictionary.
+- `return jsonify({"task": task})` returns a JSON response containing the updated task.
+
+### DELETE Request
+A DELETE request is used to remove a resource from a server. To delete a task, implement a route that handles DELETE requests:
+```python
+@app.route("/tasks/<int:task_id>", methods=["DELETE"])
+def delete_tasks(task_id):
+    task = next((item for item in tasks if item["id"] == task_id), None)
+    if task is None:
+        return jsonify({"error": "Task not found"}), 404
+    tasks.remove(task)
+    return jsonify({"result": "Task deleted"}), 200
+```
+- `@app.route` binds a URL pattern `/tasks/<int:task_id>` to the function that follows it.
+- `<int:task_id>` is a variable part of the URL that will be captured and passed to the function as an argument.
+- `methods=["DELETE"]` specifies that this route should respond to the HTTP DELETE requests. This means that when a client sends a DELETE request to the `/tasks/<int:task_id>` URL, Flask will call the associated function.
+- `delete_task()` function will take `task_id` as the argument that is extracted from `/tasks/<int:task_id>` URL. The function will then search for a task with that ID in the `tasks` list. If a task is found, it will remove the task from the list and return a JSON response indicating that the task was deleted. If no task is found, it will return a `404` status code with an error message.
+- `tasks.remove(task)` removes the task from the `tasks` list.
+- `return jsonify({"result": "Task deleted"})` returns a JSON response indicating that the task was successfully deleted.
+
+### Running the Application
+To run your application, add the following line:
+```python
+if __name__ == "__main__":
+    app.run(debug=True)
+```
+
+Then open your terminal and run the following command from your project directory:
+```bash
+python3 app.py
+```
+Your REST API is now up and running on `http://127.0.0.1:5000`
+
```

Comment:
Including something like the entire 'app.py' file looks like this for post requests or any request will be effective and clear for the reader.

Reply:
or link to GitHub repo or similar?